src/memoryprofiler.js

```diff
@@ -485,17 +496,22 @@ var emscriptenMemoryProfiler = {
     html += '. STACK_MAX: ' + toHex(STACK_MAX, width) + '.';
     html += '<br />STACK memory area used now (should be zero): ' + self.formatBytes(STACKTOP - STACK_BASE) + '.' + colorBar('#FFFF00') + ' STACK watermark highest seen usage (approximate lower-bound!): ' + self.formatBytes(Math.abs(self.stackTopWatermark - STACK_BASE));
 
-    var DYNAMIC_BASE = {{{ getQuoted('DYNAMIC_BASE') }}};
-    // During startup sbrk may not be defined yet. Ideally we should probably
-    // refactor memoryprofiler so that it only gets here after compiled code is
-    // ready to be called. For now, if the runtime is not yet initialized,
-    // assume the brk is right after the stack.
-    var DYNAMICTOP = runtimeInitialized ? _sbrk() : STACK_BASE;
-    html += "<br />DYNAMIC memory area size: " + self.formatBytes(DYNAMICTOP - DYNAMIC_BASE);
-    html += ". DYNAMIC_BASE: " + toHex(DYNAMIC_BASE, width);
-    html += ". DYNAMICTOP: " + toHex(DYNAMICTOP, width) + ".";
-    html += "<br />" + colorBar("#6699CC") + colorBar("#003366") + colorBar("#0000FF") + "DYNAMIC memory area used: " + self.formatBytes(self.totalMemoryAllocated) + " (" + (self.totalMemoryAllocated * 100 / (HEAP8.length - DYNAMIC_BASE)).toFixed(2) + "% of all dynamic memory and unallocated heap)";
-    html += "<br />Free memory: " + colorBar("#70FF70") + "DYNAMIC: " + self.formatBytes(DYNAMICTOP - DYNAMIC_BASE - self.totalMemoryAllocated) + ", " + colorBar('#FFFFFF') + 'Unallocated HEAP: ' + self.formatBytes(HEAP8.length - DYNAMICTOP) + " (" + ((HEAP8.length - DYNAMIC_BASE - self.totalMemoryAllocated) * 100 / (HEAP8.length - DYNAMIC_BASE)).toFixed(2) + "% of all dynamic memory and unallocated heap)";
+    if (runtimeInitialized) {
+      // During startup sbrk may not be defined yet. Ideally we should probably
+      // refactor memoryprofiler so that it only gets here after compiled code is
+      // ready to be called. For now, if the runtime is not yet initialized,
+      // assume the brk is right after the stack.
+      var heap_base = Module['___heap_base'];
```

Comment:
Instead of exporting and reading `__heap_base`, how about calling `sbrk()` during startup and saving that value to a global `HEAP_BASE` (or even keep the name `DYNAMIC_BASE` if that's simpler)?

Using `addAtInit` can add code to run when the runtime is initializing, and before global ctors or anything else that could use `sbrk`.

Reply:
I think the best thing to do here would be to avoid calling updateUI at all until after runtimeInitialized..  this is basically what the comment here says, but I'd rather leave that to a followup since this existing working works just fine..

Reply:
Sorry I have the base branch set wrong... think PR is just removing a couple of lines of code.

Reply:
Sorry about that.. trying to stack changes in github is hard/annoying.